feat: add emerging markets rule

vitalymatyushik opened the pull request:
Add "Emerging Markets" rule

[Related issue ](https://github.com/ghostfolio/ghostfolio/issues/3930)

![image](https://github.com/user-attachments/assets/85123acf-bc4c-4747-b69b-7b09935ba507)

dtslvr commented on apps/api/src/app/portfolio/portfolio.service.ts:
```diff
@@ -1180,6 +1181,11 @@ export class PortfolioService {
                     this.exchangeRateDataService,
                     accounts
+                  ),
+                  new AllocationClusterRiskEmergingMarkets(
```
Move to a new section:
`allocationClusterRisk: summary.ordersCount > 0 ? await this.rulesService.evaluate([ ... ])`

dtslvr commented on apps/api/src/app/portfolio/portfolio.service.ts:
```diff
@@ -1160,7 +1161,7 @@ export class PortfolioService {
     const userSettings = <UserSettings>this.request.user.Settings.settings;
 
-    const { accounts, holdings, summary } = await this.getDetails({
+    const { accounts, holdings, summary, markets } = await this.getDetails({
```
Reorder alphabetically please:
`const { accounts, holdings, markets, summary } = ...`

dtslvr commented on apps/api/src/app/user/user.service.ts:
```diff
@@ -215,6 +216,10 @@ export class UserService {
         {}
       ).getSettings(user.Settings.settings),
+      AllocationClusterRiskEmergingMarkets:
+        new AllocationClusterRiskEmergingMarkets(undefined, 0, 0).getSettings(
```
`undefined, undefined, undefined`

dtslvr commented on apps/api/src/models/rules/allocation-cluster-risk/emerging-markets.ts:
```diff
@@ -0,0 +1,84 @@
+          ruleSettings.thresholdMax * 100
+        ).toPrecision(
+          3
+        )}% of your initial investment (${(emergingMarketsValueRatio * 100).toPrecision(3)}%)`,
```
of your **current** investment

dtslvr commented on apps/api/src/models/rules/allocation-cluster-risk/emerging-markets.ts:
```diff
@@ -0,0 +1,84 @@
+          ruleSettings.thresholdMin * 100
+        ).toPrecision(
+          3
+        )}% of your initial investment (${(emergingMarketsValueRatio * 100).toPrecision(3)}%)`,
```
of your **current** investment

dtslvr reviewed (changes requested):
Thanks for this contribution, @vitalymatyushik! 😎

As this is the first rule with `thresholdMin` and `thresholdMax`, it is a bit more tricky. I have added a few comments and hope that clarifies it.

vitalymatyushik commented:
thank you @dtslvr!
I've addressed the comments and added a new section 

<img width="1132" alt="image" src="https://github.com/user-attachments/assets/63948923-7da5-4cc5-a524-abe42689ee9a">

dtslvr reviewed (approved):
This is a great addition! 🙌🏻 Thank you, @vitalymatyushik!
Ready to merge.
